splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/joins/CrossJoinIT.java

```diff
@@ -328,7 +328,8 @@ public void testPickBroadcastEquality() throws Exception {
                 format("explain select count(*) from %s as a inner join %s as b " +
                         "--SPLICE-PROPERTIES useSpark=%s \n" +
                         " on a.a1 = b.a1" , s1, s1, useSpark);
-        rowContainsQuery(6, sqlText,"Broadcast", classWatcher);
+        String expected = useSpark ? "CrossJoin" : "Broadcast";
```

Comment:
@ascend1 Can you check further why we pick cross join over broadcast join? With equality join condition, we should pick pick broadcast join instead of cross join. Thanks!

Reply:
Yes, I debugged into this test and its about the change on outerCost.RemoteCost. Not adding it to the local cost making the total cost of cross join lower.

I then read some articles about Spark broadcast join and cross join. Now I see that broadcast join is not only about broadcasting but also hashing the smaller side. Now it makes total sense that we should use it for equality conditions.

I revert the change in crossJoinStrategyLocalCost and this test. Now the optimizer should make the same choice as before.

Reply:
Also, it seems after this change: https://github.com/splicemachine/spliceengine/pull/3826/files, `UpgradeScriptToInvalidateStoredStatement` becomes a default upgrade action.

Reply:
You are right, I forgot about the change of DB-9821 which removes the need to add the upgrade script. Thanks!